ui/actions/nuxeo-share-button.js

```diff
@@ -136,28 +148,39 @@ import './nuxeo-action-button-styles.js';
 
     _toggleDialog() {
       this.$.dialog.toggle();
-      this._selectPermalink();
     }
 
     _buildPermalink(document) {
       return document ? `${window.location.origin + window.location.pathname}#!/doc/${document.uid}` : '';
     }
 
-    _copyPermalink() {
-      this._selectPermalink();
+    _copyLink(e) {
+      const button = e.currentTarget;
+      const link = button.previousElementSibling;
+
+      this._selectLink(link);
       if (!window.document.execCommand('copy')) {
         return;
       }
 
-      const link = this.$.copyLink;
-      this._debouncer = Debouncer.debounce(this._debouncer, timeOut.after(3000), () => {
-        link.innerText = this.i18n('shareButton.operation.copy');
+      button._debouncer = Debouncer.debounce(button._debouncer, timeOut.after(2000), () => {
+        this._unselectLink(link);
+        button.set('icon', 'link');
+        button.classList.remove('selected');
       });
-      link.innerText = this.i18n('shareButton.operation.copied');
+
+      button.set('icon', 'check');
+      button.classList.add('selected');
+      this.fire('notify', { message: this.i18n('shareButton.operation.copied'), duration: 2000 });
+    }
+
+    _selectLink(link) {
```

Comment:
No need to extract a function that is only used once? (and in this case, only has a single line of code)